processor/processor.go

```diff
@@ -700,6 +700,33 @@ func makeCommonMetadataFromSingularEvent(singularEvent types.SingularEventT, bat
 	return &commonMetadata
 }
 
+func makeCommonMetadataFromSingularEventWithoutMarshalling(singularEvent types.SingularEventT, batchEvent *jobsdb.JobT, receivedAt time.Time, source backendconfig.SourceT) *transformer.MetadataT {
```

Comment:
Do we need 2 methods? Why not updating the original method?

Reply:
No we don't need it.
Just left it for now for facilitating the benchmark test.
Will be removed before marking ready for review.